### PR TITLE
Macro beholder events and more careful handling of macro names, resolves SYN-7115 SYN-7184 SYN-7196

### PR DESCRIPTION
**Pull request by rakuy0**

### CODE REVIEW

File: synapse/cortex.py [hunk 1]
```diff
@@ -1151,9 +1155,15 @@ async def delStormMacro(self, name, user=None):
...
     async def _delStormMacro(self, name):
+        if not name:
+            raise s_exc.BadArg(mesg=f'Macro names must be at least 1 character long')
+
         byts = self.slab.pop(name.encode(), db=self.macrodb)
+
         if byts is not None:
-            return s_msgpack.un(byts)
+            macro = s_msgpack.un(byts)
+            await self.feedBeholder('storm:macro:del', {'name': name, 'iden': macro.get('iden')})
```
[the remark this answers is not in the document]
Comment (rakuy0) [reply synapse/cortex.py:1165]: At least for del I'm not sure what we'd get out of sending more? The only other thing that might be helpful to send is the perms, but even then that gets sent up in other messages, so it's kinda not strictly necessary. It's why most of the other feedBeholder del calls also only send the iden.

mod we're a little split on. Some calls send up the entire def, some send up the necessary edits. I personally like only sending up what's necessary, but it's not a hill I particularly feel like dying on if someone else feels strongly about it.

[the remark this answers is not in the document]
Comment (invisig0th) [reply synapse/cortex.py:1165]: It mostly just feels a bit more future proof ( classic perf / extensibility trade off conundrum :D ) for instances in the future when we realize we want some other property from the macro or some new property gets added that would be good to have in the event. I'm fine with either way :)

[the remark this answers is not in the document]
Comment (rakuy0) [reply synapse/cortex.py:1165]: For mod it probably makes sense and I can add that in. 

If we want to expand what del does, I'd kinda wanna do it for all of the other feedBeholder calls too so they're all consistent, which would force a bunch of other changes in optic.

File: synapse/lib/stormlib/macro.py [hunk 2]
```diff
@@ -74,7 +74,7 @@
         'storm': '''
             $count = $(0)
             for ($name, $mdef) in $lib.macro.list() {
-                $user = $lib.auth.users.get($mdef.user)
+                $user = $lib.auth.users.get($mdef.creator)
```
Comment (vEpiphyte) [inline synapse/lib/stormlib/macro.py:77]: If the user was deleted, this would return None and print user=$lib.null

**Review by invisig0th — APPROVED** [review]

**Pull request merged**